File: top-interview-questions-medium/Math/happy-number/202.py
class Solution:
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        set = []
        while (1):
            n = self.helper(n)
            if n == 1: return True
            if n in set: return False
            set.append(n)

    def helper(self, s):
        s = str(s)
        val = 0
        for i in list(s):
            val += int(i) ** 2
        return val

File: top-interview-questions-medium/Math/happy-number/202.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
  const set = []

  const helper = s => {
    s = String(s)
    let val = 0
    for (let i of s.split('')) {
      val += Number(i) ** 2
    }
    return val
  }

  while (true) {
    n = helper(n)
    if (n === 1) return true
    if (set.includes(n)) return false
    set.push(n)
  }
}
